### ISSUE TITLE
Media & Text block: radius doesn't apply to media element and goes underneath

### ISSUE THREAD

**Issue by Robertght** [WordPress/gutenberg#69596]
### Description

The border radius doesn't go on top of the existing content, but underneath it.

### Step-by-step reproduction instructions

1. Create a new page
2. Add a media & text block
3. Set a color to the border so it's more obvious
4. Increase that radius until you can see the overlap

### Screenshots, screen recording, code snippet

Editor:

![Image](https://github.com/user-attachments/assets/668f5eba-21f1-43a2-b9a8-58589f0f6184)

Frontend:

![Image](https://github.com/user-attachments/assets/4c65a081-fb97-44cd-80b2-6b12af72c176)

### Environment info

Gutenberg 20.4.0
WordPress 6.7.2

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by Mayank-Tripathi32** [comment]
Hi @Robertght, Thank you for reporting the issue.

I think adding `"overflow: hidden"` to `.wp-block-media-text`  will resolve the issue but it might cause unexpected changes. I will open a PR to investigate futher.


Here is clip with overflow hidden via inspector
https://q7utzrengv.ufs.sh/f/Wgl9eBAmTj29Wijoz5AmTj29whDvJ487kFt0Sz6RbdQVMYrN

**Comment by Mayank-Tripathi32** [comment]
I see that @Debarghya-Banerjee has opened a similar PR. Will be testing that instead. Thanks

**Comment by t-hamano** [comment]
I think adding `overflow: hidden` causes the a11y/UX problems. It may becomes difficult to select text and inner blocks.

See:

- https://github.com/WordPress/gutenberg/pull/63428#issuecomment-2227963763
- https://github.com/WordPress/gutenberg/pull/63542#pullrequestreview-2179317397

**Comment by Mayank-Tripathi32** [comment]
> Ultimately, I think refactoring this block to use an inner Image block, is our best bet. It would then allow optional control over the inner image border-radius.

I think we can pursue this for resolving the issue? It will allow user to set radius on image itself and give more freedom over image?

**Comment by t-hamano** [comment]
>Ultimately, I think refactoring this block to use an inner Image block, is our best bet.

Isn't this the same as the Columns block or Row block?

> It would then allow optional control over the inner image border-radius.

I'm not sure this control covers all scenarios. For example, we would expect the image to have rounded corners on the top left and bottom left. However, if the media and content are vertically stacked, we would expect the image to have rounded corners on the top left and top right.

We probably need a new responsive UI to solve this issue with an ideal approach.

References:

- https://github.com/WordPress/gutenberg/issues/34641
- https://github.com/WordPress/gutenberg/issues/19909

**Comment by Mayank-Tripathi32** [comment]
> I'm not sure this control covers all scenarios. For example, we would expect the image to have rounded corners on the top left and bottom left. However, if the media and content are vertically stacked, we would expect the image to have rounded corners on the top left and top right.

That's true. We need a way to differentiate between mobile, tablet, and desktop views to ensure the correct styling, even after adding an inner block.
In the meantime, should we explore a temporary fix?

**Comment by t-hamano** [comment]
> In the meantime, should we explore a temporary fix?

What kind of temporary fix do you have in mind?

**Comment by Mayank-Tripathi32** [comment]
> > In the meantime, should we explore a temporary fix?
> 
> What kind of temporary fix do you have in mind?

I was thinking of applying a border-radius to the image’s left and top in a horizontal layout, and only to the top when stacked vertically. This would temporarily solve the issue. We could also add a small toggle so users can choose whether the image should inherit the radius.

**Comment by t-hamano** [comment]
> I was thinking of applying a border-radius to the image’s left and top in a horizontal layout, and only to the top when stacked vertically. This would temporarily solve the issue. We could also add a small toggle so users can choose whether the image should inherit the radius

It is important to note that newly added features and APIs cannot be easily removed, even if it is a temporary fix. We always need to strive to maintain backward compatibility.

**Comment by Mayank-Tripathi32** [comment]
> It is important to note that newly added features and APIs cannot be easily removed, even if it is a temporary fix. We always need to strive to maintain backward compatibility.

True, I guess we will have to wait on it in that case.

**Comment by NidhiDhandhukiya74** [comment]
Hello @t-hamano & @Mayank-Tripathi32 
I have added the PR as per your above concerns.
To ensure backward compatibility, border-radius applies only when set by the user, preventing unintended changes to existing blocks. The solution avoids fixed values, using inherit for flexibility and proper responsive rendering without breaking past implementations.

**Comment by t-hamano** [comment]
@NidhiDhandhukiya74 Thanks for submitting the issue.

> I have added the PR as per your above concerns.

As far as I know, I think there is no perfect solution to this.

- Users may have added their own media queries, forcing vertical layouts at breakpoints other than 768px.
- When users apply the border-radius styles via the Global Styles, Media & Text Blocks don't have inline styles, so we can't use the selector depending on `[style*="border-radius"]`.

Personally, I would be wary of temporary fixes until we find an approach that ideally solves this problem.

**Comment by djcowan** [comment]
<del>
A potential progressive enhancement path might be possible through the use of data-attributes and inheritence. 
A very quick example ( Note# : the example includes a defined unit length, which is not ideal but I was unable to get the prototype working in chrome 136.0.7103.114 (Official Build) (64-bit) without the unit. A `raw-string` value like `16px` or `1rem`  had no effect.


 `// data-radius="<value>"`


```html
<div class="wp-block-media-text has-media-on-the-right" style="border-radius:2rem" data-radius="15">
	<div class="wp-block-media-text__content">
		<h3 class="wp-block-heading">Media and text</h3>
	</div>
	<figure class="wp-block-media-text__media">
		<img width="1600" height="1067" alt="" src="https://wpthemetestdata.files.wordpress.com/2013/03/image-alignment-1200x4002.jpg">
	</figure>
</div>
```

```css

/** via a css variable */

.wp-block-media-text:where([data-radius]) {
	--prefix--data--radius: attr(data-radius px, initial);
}

.wp-block-media-text:where([data-radius]) > .wp-block-media-text__media {
    clip-path: inset(1px 1px 1px 1px round var(--prefix--data--radius));
}

```
</del>

^ I must have been very tired when I wrote this. There are far better approaches available and in the discussion pipeline. As illustrated by @kizu [kizu](<https://kizu.dev/>) and outlined: [css-borders-4](<https://github.com/w3c/csswg-drafts/issues/7707>), [self()](<https://github.com/w3c/csswg-drafts/issues/9459>)

**Comment by scruffian** [comment]
This one seems connected to https://github.com/WordPress/gutenberg/issues/45353. Maybe the solution is for the border control to apply clipping like we do in the cover block.
